public/jsx/components/card-choice.jsx

```diff
@@ -64,7 +65,8 @@ module.exports = React.createClass({
     for (var i = 0; i < cardData.length; i += 1) {
       var card = cardData[i];
```

Comment:
This code runs within a render function which means it gets called for any state change. I thought of a situation where a person has multiple cards, selects one, submits it, and gets a gateway error like 'card declined'. In that case, the form will be re-rendered and ``this.state.card`` will be set to the one he selected. So, you need to do the checked feature more like this:

````
var checked = this.state.card && card.resource_uri === this.state.card;
cardList.push(<CardItem ... checked={checked} />);
````

You may want to add another test (in the multiple cards test case) to make sure the right one gets the ``checked`` property.